src/audio/smart_amp/smart_amp.c

```diff
@@ -589,56 +582,49 @@ static int smart_amp_copy(struct comp_dev *dev)
 	comp_dbg(dev, "smart_amp_copy()");
 
 	/* available bytes and samples calculation */
-	avail_passthrough_frames = audio_stream_avail_frames(&source_buf->stream,
-							     &sink_buf->stream);
+	avail_passthrough_frames = audio_stream_avail_frames(&sad->source_buf->stream,
+							     &sad->sink_buf->stream);
 
 	avail_frames = avail_passthrough_frames;
 
 	if (sad->feedback_buf) {
-		struct comp_buffer *feedback_buf = buffer_acquire(sad->feedback_buf);
```

Comment:
maybe we should keep this pointer - just remove the call to `buffer_acquire()`. This is a `.copy()` function and our recent trend has been distrusting the compiler to make simple meaningful optimisations like this one and doing them manually... Maybe for `source_buf` and `sink_buf` too. That also reduces the chance of an accidental mistake

Reply:
Does it look good now?